onnxscript/rewriter/generic_pattern.py

```diff
@@ -294,6 +294,78 @@ def make_node(
         return proto
 
 
+class PatternMatchResult:
```

Comment:
I like this structured way of returning results.

Reply:
This is good. Ideally, we should unify this and the other pattern-match-result [MatchBinding](https://github.com/microsoft/onnxscript/blob/e3bbfaa9e1ba91fa2e21e812fc378ce780babdac/onnxscript/rewriter/pattern.py#L278) to avoid confusion for users ... I am not saying we need to do this in this PR, but eventually. Since we have lot of concurrent work, it is ok to do things piece-meal for efficiency.

Reply:
Do we have other structured representation too? Sorry I wasn't aware. I think we should not create new structures that duplicate current structures.

Reply:
It also looks like MatchResult serves as an input to user code. PatternMatchResult is used to carry the rewritten results? In both cases I am still not sure why a user would need to initialize this object.